### PR TITLE
Add assertions to find command constructor

### LINKED ISSUE #114: Add assertions to find command

### PR DESCRIPTION
**Pull request by Vatinius**
Fixes #114 . Also fixed minor formatting issue in AddressBookParser javadocs.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 1]
```diff
@@ -37,12 +37,12 @@ public class AddressBookParser {
     private static final Logger logger = LogsCenter.getLogger(AddressBookParser.class);
 
     /**
-     * Used to keep track of all the commands available
+     * Used to keep track of all the commands available.
```
Comment (tayxuenye) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:40]: Good attention to detail

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ public class FindCommand extends Command {
      *                         a predicate which returns true only when all the keywords match, is loaded.
      */
     public FindCommand(ContainsSpecificKeywordsPredicate specificPredicate) {
-        requireNonNull(specificPredicate);
+        assert specificPredicate != null;
```
Comment (tayxuenye) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:56]: Ok, simple but important assertion

**Review by tayxuenye — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 2 above)
Comment (sayomaki) [reply src/main/java/seedu/address/logic/commands/FindCommand.java]: Technically the `requireNonNull` function already enforces that the value should be not null... and it also runs at runtime too

Comment (Vatinius) [reply src/main/java/seedu/address/logic/commands/FindCommand.java]: > Technically the `requireNonNull` function already enforces that the value should be not null... and it also runs at runtime too

Touche. I did think about this too lol. I'll try find somewhere else that i can put assertions instead.

**Review by BuffWuff1712 — APPROVED** [review]
lgtm

**Pull request merged**